File: 2026/08-August/30-Sunday/solution.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();

        int minIdx = 0, maxIdx = 0;

        // Find indices of minimum and maximum
        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[minIdx])
                minIdx = i;

            if (nums[i] > nums[maxIdx])
                maxIdx = i;
        }

        // Ensure minIdx comes before maxIdx
        if (minIdx > maxIdx)
            swap(minIdx, maxIdx);

        // 1. Remove both from the front
        int front = maxIdx + 1;

        // 2. Remove both from the back
        int back = n - minIdx;

        // 3. Remove min from front and max from back
        int both = (minIdx + 1) + (n - maxIdx);

        return min({front, back, both});
    }
};

File: 2026/08-August/30-Sunday/solution.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)

        # Find indices of minimum and maximum
        min_idx = nums.index(min(nums))
        max_idx = nums.index(max(nums))

        # Make min_idx the smaller index
        if min_idx > max_idx:
            min_idx, max_idx = max_idx, min_idx

        # 1. Remove both from the front
        front = max_idx + 1

        # 2. Remove both from the back
        back = n - min_idx

        # 3. Remove one from front and one from back
        both = (min_idx + 1) + (n - max_idx)

        return min(front, back, both)
